libjolie/src/main/java/jolie/lang/parse/AbstractParser.java

```diff
@@ -180,62 +180,120 @@ protected final void setScanner( Scanner scanner ) {
 
 
 	/**
-	 * Returns the current {@link ParsingContext} from the underlying {@link Scanner}
+	 * Returns a simple {@link ParsingContext} from the underlying {@link Scanner} and does not take
+	 * startLine and endLine into account
 	 * 
 	 * @return the current {@link ParsingContext} from the underlying {@link Scanner}
 	 */
 	public final ParsingContext getContext() {
-		return new URIParsingContext( scanner.source(), scanner.startLine(), scanner.endLine(), scanner.errorColumn(),
+		return new URIParsingContext( scanner.source(), scanner.line(), scanner.line(), scanner.errorColumn(),
+			scanner.errorColumn() + token.content().length(),
 			scanner.codeLine() );
 	}
 
 	/**
-	 * Returns the current {@link ParsingContext} from the underlying {@link Scanner}
+	 * Returns the current {@link ParsingContext} from the underlying {@link Scanner} taking start and
+	 * end line into account
 	 * 
 	 * @return the current {@link ParsingContext} from the underlying {@link Scanner}
 	 */
 	public final ParsingContext getContextDuringError() throws IOException {
-		int linenr = scanner.line();
+		// read the rest of the line, so we can use the whole line in the error message
 		readLineAfterError();
-		if( linenr < scanner.line() ) {
-			return new URIParsingContext( scanner.source(), scanner.startLine(), scanner.endLine(),
-				scanner.errorColumn(),
-				scanner.codeLine() );
-		} else if( scanner.errorColumn() < 0 ) {
-			String correctLine = scanner.getAllCodeLines().get( linenr - 1 );
-			if( !correctLine.endsWith( "\n" ) ) {
-				correctLine += "\n";
+		// If startline, endline and line are 1, the line is 1. If startline and endline are 1, but line is
+		// not then startline and endline have not been set yet and line is line
+		if( scanner.startLine() == 0 && scanner.endLine() == 0 && scanner.line() >= 0 ) {
```

Comment:
What does it sound to you to have a state, or just an utility function on the scanner, so we don't have to worry much about the comparison logic on this function.

Reply:
So a function which checks the scanner lines and returns a kind of state, which I then use in this function instead of doing the comparisons here? I could do that.